File: 0004_median_of_two_sorted_arrays/Solution.js
var findMedianSortedArrays = function (nums1, nums2) {
    let lgth1 = nums1.length, lgth2 = nums2.length, lgth3 = lgth1 + lgth2;
    let mergedArr = [];
    let i = 0, j = 0;

    while (i < lgth1 && j < lgth2) {
        if (nums1[i] < nums2[j]) mergedArr.push(nums1[i++]);
        else mergedArr.push(nums2[j++]);
    }

    while (i < lgth1) {
        mergedArr.push(nums1[i++]);
    }

    while (j < lgth2) {
        mergedArr.push(nums2[j++]);
    }

    if (lgth3 % 2 != 0) return mergedArr[parseInt(lgth3 / 2)];

    return (mergedArr[parseInt(lgth3 / 2)] + mergedArr[parseInt((lgth3 / 2) - 1)]) / 2;
};

File: 0004_median_of_two_sorted_arrays/Solution.java
// Return the median of the two sorted arrays
// Completed on 05/02/2024 (Last update on 24/02/2024)

class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int sz1 = nums1.length, sz2 = nums2.length, sz3 = sz1 + sz2;
        int[] mergedArr = new int[sz3];
        int i = 0, j = 0, k = 0;

        while (i < sz1 && j < sz2) {
            if (nums1[i] < nums2[j]) mergedArr[k++] = nums1[i++];
            else mergedArr[k++] = nums2[j++];
        }

        while (i < sz1) {
            mergedArr[k++] = nums1[i++];
        }

        while (j < sz2) {
            mergedArr[k++] = nums2[j++];
        }

        if (sz3 % 2 != 0) return (double) mergedArr[sz3 / 2];

        return (double) (mergedArr[(sz3 / 2) - 1] + mergedArr[sz3 / 2]) / 2;
    }
}
